rfcs/0006-vesting-schedules.md

```diff
@@ -0,0 +1,94 @@
+Design Proposal: Pablo Fees & Staking Rewards Distribution
+==========================================================
+
+Table of Contents
+
+- [1. Abstract](#1-abstract)
+- [2. Background](#2-background)
+- [3. Use Cases](#3-requirements)
+- [4. Method](#4-method)
+  - [4.1. Structs](#41-structs)
+  - [4.2. Storage](#42-storage)
+  - [4.3. Events](#43-events)
+  - [4.4. Errors](#44-errors)
+  - [4.5. Extrinsics](#45-extrinsics)
+    - [4.5.1 vested_transfer](#451-vested_transfer)
+    - [4.5.2 claim](#452-claim)
+
+## 1. Abstract
+
+This document proposes that vesting schedules are tracked individually, instead of being tracked as a group based on their asset id.
+
+## 2. Background
+
+Currently, the `vesting` pallet groups vesting schedules by `asset id`. This means that if a user has multiple
+vesting schedules for the same asset, they will all be grouped together, and there will be no distinction between
+them. Therefore, when a user wants to `claim`, it is only possible to claim **everything** that has been vested for
+a given asset. We need to change this in order to allow users to claim only a chosen vesting schedule.
+
+## 3. Requirements
+
+The vesting pallet
+
+1. MUST keep track of individual vesting schedules for each asset.
+2. MUST allow a user to choose which vesting schedule to claim.
+3. MUST include the claimed schedule in the corresponding event.
+4. MUST not do anything when a non-existent vesting schedule is claimed.
+5. SHOULD allow users to claim **all** vesting schedules for the same asset, by not specifying a `vesting_schedule_id`
```

Comment:
```suggestion
```
Existing functionality so not mentioning it will imply the functionality will remain the same